core/kazoo_number_manager/src/knm_port_request.erl

```diff
@@ -563,6 +594,68 @@ transition_numbers(PortReq) ->
             end
     end.
 
+update_used_by(_Numbers, 'undefined') -> ok;
+update_used_by(Numbers, App) ->
+    lager:debug("update number ~p with used_by ~p", [Numbers, App]),
+    knm_numbers:assign_to_app(Numbers, App).
+
+-spec app_used_by_portin(kz_term:ne_binary(), kz_json:object()) -> 'undefined' | kz_json:object().
+app_used_by_portin(Number, JObj) ->
+    Numbers = kzd_port_requests:numbers(JObj),
+    kz_json:get_value([Number, ?USED_BY_KEY], Numbers).
+
+maybe_reconcile_app_used_by(Num, AccountId, JObj) ->
```

Comment:
Instead query db for each number you can just simply use used `[{'keys', Numbers}]` as view option.

So what you have to change here is:
1. create a proplist of form `[{Number, Appname}]` from port's numbers field
1. use `keys` view to to query dbs
1. check query db against the numbers in step one above and create a list of app assignment for each number if they have any
1. call `assign_to_app`